### DISCUSSION TITLE
About Sync on appflowy

### DISCUSSION THREAD

**Discussion by huabin** [AppFlowy-IO/AppFlowy#183] [Technical help wanted]
Hey guys,

I checked appflowy roadmap on trello.com, and I would like to ask what way it will be implemented? Will it use a third-party storage drive like Google drive, or will appflowy have a built-in sync service, or will it allow users to set up a hosting of their own?

 I would prefer users could use self-hosting.

Thank you.

**Comment by annieappflowy** [comment]
We plan to enable ppl to self-host and support syncing across devices. Third-party storage options would be open to people who can write code. We're open to suggestions.

**Reply by huabin** [reply]
A little more detail?

**Reply by MikeWallaceDev** [reply]
@appflowy and I are looking at the design of the application so that it can work both as a web application and as a standalone application. This requires some big decisions on where we are going to place all the components of the design. The goal is to design the application in such a way as to allow for any type of storage (by using interfaces, components, and proxy classes). Once we have this done, then it's just a matter of creating the component for the desired storage. The application won't care about the type of storage.

As for which storage will be implemented? This is an open source project, so any developer will be able to create a component to connect to whatever data source they like. As for which data sources will be implemented by the core team, that has not been decided yet.

**Reply by huabin** [reply]
Got it.
